src/server/api/routers/generations.ts

```diff
@@ -88,92 +88,99 @@ export const generationsRouter = createTRPCRouter({
-            ${searchCondition}
-            ${filterCondition}
-          ORDER BY o.start_time DESC
-          LIMIT ${input.limit}
-          OFFSET ${input.page * input.limit}
-        `,
-      );
-
-      const totalGenerations = await ctx.prisma.$queryRaw<
-        Array<{ count: bigint }>
-      >(
-        Prisma.sql`
-          WITH observations_with_latency AS (
-            SELECT
-              o.*,
-              CASE WHEN o.end_time IS NULL THEN NULL ELSE (EXTRACT(EPOCH FROM o."end_time") - EXTRACT(EPOCH FROM o."start_time"))::double precision END AS "latency"
-            FROM observations o
-            WHERE o.type = 'GENERATION'
-            AND o.project_id = ${input.projectId}
-            ${datetimeFilter}
-          )
+            o.*,
+            CASE WHEN o.end_time IS NULL THEN NULL ELSE (EXTRACT(EPOCH FROM o."end_time") - EXTRACT(EPOCH FROM o."start_time"))::double precision END AS "latency"
+          FROM observations o
+          WHERE o.type = 'GENERATION'
+          AND o.project_id = ${input.projectId}
+          ${datetimeFilter}
+        ),
+        -- used for filtering
+        scores_avg AS (
           SELECT
-            count(*)
-          FROM observations_with_latency o
-          JOIN traces t ON t.id = o.trace_id
-          WHERE
-            t.project_id = ${input.projectId}
-            ${searchCondition}
-            ${filterCondition}
-        `,
+            trace_id,
+            jsonb_object_agg(name::text, avg_value::double precision) AS scores_avg
+          FROM (
+            SELECT
+              trace_id,
+              name,
+              avg(value) avg_value
+            FROM
+              scores
+            GROUP BY
+              1,
+              2
+            ORDER BY
+              1) tmp
+          GROUP BY
+            1
+        )
+        SELECT
+          o.id,
+          o.name,
+          o.model,
+          o.start_time as "startTime",
+          o.end_time as "endTime",
+          o.latency,
+          o.input,
+          o.output,
+          o.metadata,
+          o.trace_id as "traceId",
+          t.name as "traceName",
+          o.completion_start_time as "completionStartTime",
+          o.prompt_tokens as "promptTokens",
+          o.completion_tokens as "completionTokens",
+          o.total_tokens as "totalTokens",
+          o.level,
+          o.status_message as "statusMessage",
+          o.version,
+          (count(*) OVER())::int AS "totalCount"
+        FROM observations_with_latency o
+        JOIN traces t ON t.id = o.trace_id
+        LEFT JOIN scores_avg AS s_avg ON s_avg.trace_id = t.id
+        WHERE
+          t.project_id = ${input.projectId}
+          ${searchCondition}
+          ${filterCondition}
+        ORDER BY o.start_time DESC
+        LIMIT ${input.limit}
+        OFFSET ${input.page * input.limit}
+      `,
       );
 
+      const scores = await ctx.prisma.score.findMany({
+        where: {
```

Comment:
to be safe, can we add a projectId condition here to ensure, only scores from the right project are selected?